service/history/ndc/events_reapplier.go

```diff
@@ -96,13 +96,16 @@ func (r *EventsReapplierImpl) ReapplyEvents(
 		return nil, serviceerror.NewInternal("unable to reapply events to closed workflow.")
 	}
 
+	shouldScheduleWorkflowTask := false
```

Comment:
@yux0 I found some [duplicated logic](https://github.com/temporalio/temporal/blob/master/service/history/api/reapplyevents/api.go#L182) for scheduling a new workflow task in the api handler for ReapplyEvents. To me it seems like this eventsReapplier already takes care of scheduling new workflow task so in the api handler, we should always use false for the `CreateWorkflowTask` field of update workflow post action.

Reply:
Synced with @yux0 offline, we should removed the above-linked duplicated logic for scheduling new workflow task in ReapplyEvents api handler.
@pdoerner you can decide if you want to do this in a separate PR.